src/stores/right-panel/RightPanelStore.ts

```diff
@@ -230,11 +230,19 @@ export default class RightPanelStore extends ReadyWatchingStore {
     }
 
     private emitAndUpdateSettings() {
+        if (this.global?.history) {
+            this.global.history = this.global?.history.filter((card)=>this.isCardStateValid(card));
+        }
         const storePanelGlobal = convertToStorePanel(this.global);
         SettingsStore.setValue("RightPanel.phasesGlobal", null, SettingLevel.DEVICE, storePanelGlobal);
 
         if (!!this.viewedRoomId) {
-            const storePanelThisRoom = convertToStorePanel(this.byRoom[this.viewedRoomId]);
+            const panelThisRoom = this.byRoom[this.viewedRoomId];
+            if (panelThisRoom?.history) {
+                // remove all right panel cards that would fail to render (result in a soft crash)
+                panelThisRoom.history = panelThisRoom?.history.filter((card)=>this.isCardStateValid(card));
```

Comment:
```suggestion
                panelThisRoom.history = panelThisRoom?.history.filter((card) => this.isCardStateValid(card));
```

Reply:
Can this be made a style rule? I do this mistake all the time (sorry)

Reply:
Good idea, feel free to add it to the eslint plugin https://github.com/matrix-org/eslint-plugin-matrix-org